core/src/unprocessed_transaction_storage.rs

```diff
@@ -438,33 +438,19 @@ impl ThreadLocalUnprocessedPackets {
                 .chunks(batch_size)
                 .into_iter()
                 .flat_map(|packets_to_process| {
-                    let packets_to_process = packets_to_process.into_iter().collect_vec();
-
-                    // Vec<bool> of same size of `packets_to_process`, each indicates
-                    // corresponding packet is tracer packet.
-                    let tracer_packet_indexes = packets_to_process
-                        .iter()
-                        .map(|deserialized_packet| {
-                            deserialized_packet
-                                .original_packet()
-                                .meta
-                                .is_tracer_packet()
-                        })
-                        .collect::<Vec<_>>();
-                    saturating_add_assign!(
-                        total_tracer_packets_in_buffer,
-                        tracer_packet_indexes
-                            .iter()
-                            .filter(|is_tracer| **is_tracer)
-                            .count()
-                    );
+                    // Only prcoess packets not yet forwarded
+                    let (packets_to_forward, tracer_packet_indexes) = self
```

Comment:
nit: could rename to `is_tracer_packet` or something more descriptive since these are not really indexes